pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -886,43 +932,30 @@ func (os *OpenStack) getVolume(diskName string) (volumes.Volume, error) {
 	return volume, err
 }
 
-func (os *OpenStack) getComputeIDbyHostname(cClient *gophercloud.ServiceClient) (string, error) {
+// getLocalComputeID tries to find ID of the OpenStack instance where
+// this code runs.
+func (os *OpenStack) getLocalComputeID(cClient *gophercloud.ServiceClient) (string, error) {
+	// try to read /var/lib/cloud/data/instance-id
+	const instanceIDFile = "/var/lib/cloud/data/instance-id"
+	idBytes, err := ioutil.ReadFile(instanceIDFile)
+	if err == nil {
+		id := string(idBytes)
+		id = strings.TrimSpace(id)
```

Comment:
Possibly, but "empty" is just one of several invalid values that might be in this file.  I'm sort of inclined to go with "If the file exists, trust it" without a clear reason not to...